### PR TITLE
Committing recipe for conda-forge release of pdffit2

### PR DESCRIPTION
**Pull request by dragonyanglong**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by dragonyanglong** [comment]
please review and merge.

**Comment by dragonyanglong** [comment]
@conda-forge-admin  ready for review!

**Comment by dragonyanglong** [comment]
@conda-forge/staged-recipes, hi please review.

**Comment by dragonyanglong** [comment]
@conda-forge/help-python ready for review.

File: recipes/diffpy.pdffit2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+# If your package is python based, we recommend using Grayskull to generate it instead:
+# https://github.com/conda-incubator/grayskull
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 14 is convenient
+# when copying and pasting from another recipe, but not really needed.
+{% set name = "simplejson" %}
```
Comment (xhochy) [inline recipes/diffpy.pdffit2/meta.yaml:8]: Please use the variables here to template the recipe.

File: recipes/diffpy.pdffit2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  # See https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python for more details.
+  # noarch: python
+  # skip: True  # [py<35]
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  # More info about selectors can be found in the conda-build docs: 
+  # https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#preprocessing-selectors
+  # script: {{ PYTHON }} -m pip install . -vv
+  preserve_egg_dir: True
```
Comment (xhochy) [inline recipes/diffpy.pdffit2/meta.yaml:40]:
```suggestion
```

File: recipes/diffpy.pdffit2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+  # entry_points:
+  #   - pdfgui = diffpy.pdfgui.applications.pdfgui:main
+
+requirements:
+  build:
+    # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
+    # Compilers are named 'c', 'cxx' and 'fortran'.
+    - {{ compiler('cxx') }}
+    - gsl {{ gsl }}
```
Comment (xhochy) [inline recipes/diffpy.pdffit2/meta.yaml:51]: Should be in host without the variable.

File: recipes/diffpy.pdffit2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  imports:
+    - diffpy
+    # - diffpy.pdffit2
+    # - diffpy.pdffit2.ipy_ext
+    # - diffpy.pdffit2.tests
+#   # For python packages, it is useful to run pip check. However, sometimes the
+#   # metadata used by pip is out of date. Thus this section is optional if it is
+#   # failing.
+#   # requires:
+#   #   - pip
```
Comment (xhochy) [inline recipes/diffpy.pdffit2/meta.yaml:74]: Please add pip check

**Review by xhochy — CHANGES_REQUESTED** [review]
Please remove the comments from the example

**Comment by dragonyanglong** [comment]
Dear @xhochy , @conda-forge/help-python thanks for your review. The yaml file is updated per your comments, please review.

File: recipes/diffpy.pdffit2/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    - pip check
+
+about:
+  home: https://github.com/diffpy/diffpy.pdffit2
+  summary: 'PDFfit2 - real space structure refinement program'
+  description: |
+    PDFgui is a software for pair distribution function simulation and
+    structure refinement, which provides a simple, fast, and easy-to-use user
+    interface for chemists, physicists, and materials scientists.
+  license: Modified BSD
```
Comment (xhochy) [inline recipes/diffpy.pdffit2/meta.yaml:41]: Can you please fix the linter comment? It looks like this is BSD-3-clause

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by dragonyanglong** [comment]
Thanks @xhochy , I updated it. Please review.

**Comment by dragonyanglong** [comment]
please hold on this PR, we are updating some codes of the package to be released. Once done, I will let you know. thanks.

**Comment by dragonyanglong** [comment]
@conda-forge/help-python @xhochy  this is ready for review now. Thanks so much!

**Comment by dragonyanglong** [comment]
@conda-forge/staged-recipes

File: recipes/diffpy.pdffit2/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE.txt
+  doc_url: https://www.diffpy.org/doc/pdffit2
+  dev_url: https://github.com/diffpy/diffpy.pdffit2
+
+extra:
+  recipe-maintainers:
+    - dragonyanglong
+    - sbillinge
```
Comment (xhochy) [inline recipes/diffpy.pdffit2/meta.yaml:50]: @sbillinge Please confirm that you're happy to be a maintainer here.

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by sbillinge** [comment]
I confirm

**Review by xhochy — APPROVED** [review]

**Pull request merged**